core/domain/entities/admin/GraphQLData/GeneralSettings.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+namespace EventEspresso\core\domain\entities\admin\GraphQLData;
+
+/**
+ * Class GeneralSettings
+ * Description
+ *
+ * @package EventEspresso\core\domain\entities\admin\GraphQLData
+ * @author  Manzoor Wani
+ * @since   $VID:$
+ */
+class GeneralSettings extends GraphQLData
+{
+
+    /**
+     * @param array $where_params
+     * @return array|null
+     * @since $VID:$
+     */
+    public function getData(array $where_params = [])
+    {
+        $field_key = 'generalSettings';
+        $query = <<<QUERY
+        query GET_GENERAL_SETTINGS {
+            {$field_key} {
+                dateFormat
+                timeFormat
+                timezone
+                __typename
+            }
+        }
+QUERY;
+        $data = [
+            'operation_name' => 'GET_GENERAL_SETTINGS',
+            'query'          => $query,
+        ];
+
+        if (! empty($where_params)) {
+            $data['variables'] = [];
+            $data['variables']['where'] = $where_params;
+        }
+
+        $responseData = $this->makeGraphQLRequest($data);
+        return ! empty($responseData[ $field_key ]) ? $responseData[ $field_key ] : null;
```

Comment:
`CurrentUser`, `Datetimes` and `GeneralSettings` have this block being duplicated. It's ok for 2-3 instances to have duplicate, if this will be used in more classes, than it's worth to extract somewhere.